storage/connectors/jdbc/src/test/java/feast/storage/connectors/jdbc/retriever/SnowflakeHistoricalRetrieverJSONColTest.java

```diff
@@ -0,0 +1,253 @@
+import feast.storage.api.retriever.HistoricalRetrievalResult;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+public class SnowflakeHistoricalRetrieverJSONColTest {
+  private JdbcHistoricalRetriever snowflakeFeatureRetriever;
+  //  Snowflake account
+  //  private String staging_location = System.getenv("STAGING_LOCATION");
+  private String staging_location = "s3://feast-snowflake-staging/test/";
+
+  private Map<String, String> snowflakeConfig = new HashMap<>();
+  private String SFUrl = "jdbc:snowflake://ry42518.us-east-2.aws.snowflakecomputing.com";
+  private String SFClassName = "net.snowflake.client.jdbc.SnowflakeDriver";
+  private String SFusername = System.getenv("SNOWFLAKE_USERNAME_RETRI");
+  private String SFpw = System.getenv("SNOWFLAKE_PASSWORD_RETRI");
+  private String SFDatabase = "DEMO_DB";
+  private String SFSchema = "PUBLIC";
+  private String SFTable = "FEAST_FEATURES";
+  private String SFRole = "ACCOUNTADMIN";
+  private String SFStorageIntegration = "s3_int";
+
+  @Before
+  public void setUp() {
+
+    snowflakeConfig.put("database", SFDatabase);
+    snowflakeConfig.put("schema", SFSchema);
+    snowflakeConfig.put("class_name", SFClassName);
+    snowflakeConfig.put("username", SFusername);
+    snowflakeConfig.put("password", SFpw);
+    snowflakeConfig.put("url", SFUrl);
+    snowflakeConfig.put("staging_location", staging_location);
+    snowflakeConfig.put("role", SFRole);
+    snowflakeConfig.put("storage_integration", SFStorageIntegration);
+    snowflakeConfig.put("table_name", SFTable);
+    Properties dsProperties = new Properties();
+    dsProperties.put("user", this.snowflakeConfig.get("username"));
+    dsProperties.put("password", this.snowflakeConfig.get("password"));
+    dsProperties.put("db", this.snowflakeConfig.get("database"));
+    dsProperties.put("schema", this.snowflakeConfig.get("schema"));
+    dsProperties.put("role", this.snowflakeConfig.get("role"));
+    HikariConfig hkConfig = new HikariConfig();
+    hkConfig.setMaximumPoolSize(100);
+    hkConfig.setDriverClassName(SFClassName);
+    hkConfig.setDataSourceProperties(dsProperties);
+    hkConfig.setJdbcUrl(this.snowflakeConfig.get("url"));
+    final HikariDataSource ds = new HikariDataSource(hkConfig);
+    JdbcTemplate jdbcTemplate = new JdbcTemplate(ds);
+    SnowflakeQueryTemplater snowflakeQueryTemplater =
+        new SnowflakeQueryTemplater(snowflakeConfig, jdbcTemplate);
+    snowflakeFeatureRetriever =
+        (JdbcHistoricalRetriever)
+            JdbcHistoricalRetriever.create(snowflakeConfig, snowflakeQueryTemplater);
+  }
+
+  @Test
+  public void shouldRetrieveFromSnowflakeTest3DatesWithMaxAge() {
+    String entitySourceUri = "s3://feast-snowflake-staging/test/entity_tables/entities_3dates.csv";
```

Comment:
maybe the whole URI should be configurable.  Or at least the base, so GCS can be used.  It may be necessary to upload the file as part of the test setup.

Reply:
On second thought, the above comment should apply to the IT not the unit test.  We should be mocking external dependencies here.